app/classes/robots.rb

```diff
@@ -1,18 +1,24 @@
 # frozen_string_literal: true
 
+# Permissions for user_agents
 class Robots
   class << self
-    def allowed?(args)
-      populate_allowed_robot_actions unless defined?(@@allowed_robot_actions)
-      return true  if args[:controller].start_with?("api")
-      return false if args[:ua].downcase.include?("yandex")
+    # Is the robot authorized to be on the site?
+    def authorized?(user_agent)
+      # Googlebot not followed by a hyphen excludes Googlebot-Image, etc.
+      /Googlebot(?!-)|bingbot/.match?(user_agent)
+    end
+
+    def action_allowed?(args)
+      populate_allowed_robot_actions unless defined?(@allowed_robot_actions)
+      return true if args[:controller].start_with?("api")
```

Comment:
Allows anonymous API use. 
Do we want to require a key?

Reply:
This is a question that seems to relate to the next GraphQL PR, too. 

Somehow my initial API test `HttpRequestTest` was getting identified as a robot, and I could only get it to run by modifying line 14 above with something like `.start_with?("api", "graphql")`. 

On the other hand, per consultation with Jason, we may want to set up the graphql api to require an API key regardless - and the SvelteKit front end would use its own API key. 

I don't know if the last part is strictly necessary, for example if the eventual front end is on another port of the same URL, as it is when I run the front end locally.

Reply:
Thanks @nimmolo. 
I'd like to look into this, in particular to understand it thought that request was coming from a robot.
Can you tell me how to replicate the issue?

Reply:
Hi Joe, 
I can only reproduce it on my PR branch, because it's a graphql controller test that is blocked as a robot. I can show you how over there.

But - on the other hand maybe i can resolve it another way - like giving the test request an arbitrary user agent string? Is that how we detecting robots in the first place?